### PR TITLE
feat: tests of version getter

### PR DESCRIPTION
**Pull request by AJaquier**

### CODE REVIEW

File: src/test/java/ch/heig/dil/BoomshotTest.java [hunk 1]
```diff
@@ -34,4 +35,19 @@ void output() throws Exception {
...
+        String pomContent = FilesHelper.readFromFile("pom.xml");
+        assertTrue(pomContent != null);
+        String testVers =
+                pomContent.substring(
+                        pomContent.indexOf("<version>") + 9, pomContent.indexOf("</version>"));
+        String[] rawVersion = versionProvide.getVersion();
+        String[] version = rawVersion[0].split(" ");
+        assertEquals(version[0], "Boomshot");
+
+        assertEquals(version[1], testVers);
```
Comment (vkaelin) [inline src/test/java/ch/heig/dil/BoomshotTest.java:51]:
```suggestion
        final String OPEN_TAG = "<version>";
        final String CLOSING_TAG = "</version>";
        Boomshot.VersionProvide versionProvide = new Boomshot.VersionProvide();
        String pomContent = FilesHelper.readFromFile("pom.xml");
        assertTrue(pomContent != null);
        String configVersion =
                pomContent.substring(
                        pomContent.indexOf(OPEN_TAG) + OPEN_TAG.length(), pomContent.indexOf(CLOSING_TAG));
        String[] rawVersion = versionProvide.getVersion();
        String[] generatedVersion = rawVersion[0].split(" ");
        assertEquals(generatedVersion[0], "Boomshot");

        assertEquals(generatedVersion[1], configVersion);
```

**Review by vkaelin — CHANGES_REQUESTED** [review]
Looks better good job!

Maybe take a look at the proposed changes (to remove magic numbers and some variable names).

You'll need to check if the tests pass and if you need to re-apply spotless if you accept my changes.

File: src/test/java/ch/heig/dil/BoomshotTest.java [hunk 2]
```diff
@@ -34,4 +35,21 @@ void output() throws Exception {
...
+        String pomContent = FilesHelper.readFromFile("pom.xml");
+        String configVersion =
+                pomContent.substring(
+                        pomContent.indexOf(OPEN_TAG) + OPEN_TAG.length(),
+                        pomContent.indexOf(CLOSING_TAG));
+        String[] rawVersion = versionProvide.getVersion();
+        String[] version = rawVersion[0].split(" ");
+        assertEquals(version[0], "Boomshot");
+
+        assertEquals(version[1], configVersion);
```
Comment (vkaelin) [inline src/test/java/ch/heig/dil/BoomshotTest.java:53]:
```suggestion
        final String OPEN_TAG = "<version>";
        final String CLOSING_TAG = "</version>";
        Boomshot.VersionProvide versionProvide = new Boomshot.VersionProvide();
        String pomContent = FilesHelper.readFromFile("pom.xml");
        String configVersion =
                pomContent.substring(
                        pomContent.indexOf(OPEN_TAG) + OPEN_TAG.length(),
                        pomContent.indexOf(CLOSING_TAG));
        String[] rawVersion = versionProvide.getVersion();
        String[] generatedVersion = rawVersion[0].split(" ");
        assertEquals(version[0], "Boomshot");

        assertEquals(generatedVersion[1], configVersion);
```

File: src/test/java/ch/heig/dil/BoomshotTest.java [hunk 3]
```diff
@@ -34,4 +35,21 @@ void output() throws Exception {
...
+        final String CLOSING_TAG = "</version>";
+        Boomshot.VersionProvide versionProvide = new Boomshot.VersionProvide();
+        String pomContent = FilesHelper.readFromFile("pom.xml");
+        String configVersion =
+                pomContent.substring(
+                        pomContent.indexOf(OPEN_TAG) + OPEN_TAG.length(),
+                        pomContent.indexOf(CLOSING_TAG));
+        String[] rawVersion = versionProvide.getVersion();
+        String[] generatedVersion = rawVersion[0].split(" ");
+        assertEquals(version[0], "Boomshot");
```
Comment (AJaquier) [inline src/test/java/ch/heig/dil/BoomshotTest.java:51]:
```suggestion
        assertEquals(generatedVersion[0], "Boomshot");
```

**Review by vkaelin — APPROVED** [review]
LGTM now !

**Pull request merged**